### PR TITLE
fixes #221 saves selected settings on Closure of Stage

### LINKED ISSUE #221: Clean up settings dialog
The settings dialog is currently just a draft. It should finally be finished.
One thing to think about is whether/how we implement a "save" function (currently, every change is directly written to the configuration, which get stored as soon as `PreferenceService#storeInBackground()` is called - which is currently not done by the settings dialog, AFAIK)
## Task
- Go through the settings dialog and check what needs to be improved/finished. Contact me if you have any questions
- Work out different solutions on how to save the settings (do we want a "save" button, and if so, how does it work)
- Fix what needs to be fixed

Entry point: `SettingsController` and `settings.fxml`
## Acceptance criteria
- The whole settings dialog is unit tested
- I approved the result

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/34311802-clean-up-settings-dialog?utm_campaign=plugin&utm_content=tracker%2F14200913&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F14200913&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by axel1200**
When stages closes/hides a certain event is triggered which in case of the settings window saves the changes.

### CODE REVIEW

File: src/main/java/com/faforever/client/fx/WindowController.java [hunk 1]
```diff
@@ -87,6 +89,10 @@ public void onRestoreButtonClicked() {
     restore();
   }
 
+  public void setOnHidding(EventHandler<WindowEvent> eventEventHandler) {
```
Comment (micheljung) [inline src/main/java/com/faforever/client/fx/WindowController.java]: Hiding*

**Comment by micheljung** [comment]
Code doesn't compile (see travis failure)

**Comment by micheljung** [comment]
Would you mind looking at why Travis fails?

**Comment by axel1200** [comment]
No idea what is happening here, it runs on my machine.

**Comment by micheljung** [comment]
I had this failure locally as well, I think I fixed it in my map-vault branch.

**Pull request closed without merging**